pkg/kn/flags/podspec.go

```diff
@@ -418,5 +411,12 @@ func (p *PodSpecFlags) ResolvePodSpec(podSpec *corev1.PodSpec, flags *pflag.Flag
 		}
 	}
 
+	if flags.Changed("user") {
+		err = UpdateUser(podSpec, p.User)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
Is there a reason to change the order of evaluation here?

Reply:
IMO it's very likely SC was override by `UpdateSecurityContext`.

Reply:
The order of evaluation is causing the value of the user flag to get overwritten. I have created an issue for this: https://github.com/knative/client/issues/1926